core/src/main/java/hudson/model/Computer.java

```diff
@@ -303,10 +305,12 @@ public void addAction(@NonNull Action a) {
      */
     protected @NonNull File getLogDir() {
         File dir = new File(Jenkins.get().getRootDir(),"logs/slaves/"+nodeName);
-        try {
-            IOUtils.mkdirs(dir);
-        } catch (IOException x) {
-            LOGGER.log(Level.SEVERE, "Failed to create agent log directory " + dir, x);
+        synchronized (logDirLock) {
```

Comment:
Out of curiosity: why not just a `synchronized` method? The time saved is superfluous and it is cleaner. WDYT?

Reply:
To avoid reusing the `Computer` monitor which might already be used for other purposes, and possibly introducing deadlocks or contention.